conda/dev.yml

```diff
@@ -6,7 +6,8 @@ dependencies:
   # Base
   # =================
   - pip=22.2.2
-  - python=3.9.13
+  - python=3.11
```

Comment:
The `dev.yml` constraints should align to the conda-forge [recipe](https://github.com/conda-forge/zstash-feedstock/blob/main/recipe/meta.yaml). Currently, the constraints are too restrictive with exact versions pinned.

Follow this for more guidance if needed: https://github.com/E3SM-Project/e3sm_diags/pull/1006